### PR TITLE
Issue 518 Convert ProductsCountInLayeredNavigationTest to MFTF 

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description
   This PR checks if layered navigation reflects properly the next case:
 -  1 configurable product assigned to anchored subcategory with all children disabled 

### Preconditions:
1. Create four categories assigned in ascending order:
Category 1 (non-anchor) -> Category 2 (anchor) -> Category 3 (non-anchor) -> Category 4 (anchor)
2.  Create configurable product with two configurable options and assign it to "Category 4"

### Test flow
Test1, Steps:
  1. Disable configurable options via massaction
  2. Open created non anchored categories on frontend -> On the left side in the Layered Navigation there is one category inside each of anchored with zero products inside.
  3. Open created anchored categories on frontend -> There are no products inside.

Test2, Steps:
  1. Disable configurable options via product edit page
  2. Open created non anchored categories on frontend -> On the left side in the Layered Navigation there is one category inside each of anchored with zero products inside.
  3. Open created anchored categories on frontend -> There are no products inside.

### Fixed issues
1. #518 Convert ProductsCountInLayeredNavigationTest to MFTF

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertCategoryIsEmptyActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontAssertCategoryIsEmptyActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertCategoryIsEmptyActionGroup.xml:11]: Please, rename to `AssertStorefrontCategoryIsEmptyActionGroup`

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontGoToSubCategory2PageActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontGoToSubCategory2PageActionGroup" extends="StorefrontGoToCategoryPageActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontGoToSubCategory2PageActionGroup.xml:11]: Is it necessary to navigate to a category using the navigation menu? Could we simply use an URL? I may miss something, so correct me if I'm wrong, please.

Thank you.

File: app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertAttributesQtyForShoppingOptionInLayeredNavigationActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StoreFrontAssertAttributesQtyForShoppingOptionInLayeredNavigationActionGroup">
```
Comment (rogyar) [inline app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertAttributesQtyForShoppingOptionInLayeredNavigationActionGroup.xml:11]: Please, rename as `AssertStorefrontAttributesCountForShoppingOptionInLayeredNavigationActionGroup`.

Also, I would not recommend operating the `qty` keyword in this particular case but replace it with `count` or `number`. In e-commerce, we usually use `qty`  when mentioning the number of products. So, using it for attributes might be confusing.

File: app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertProductsQtyForAttributeInLayeredNavigationActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StoreFrontAssertProductsQtyForAttributeInLayeredNavigationActionGroup">
```
Comment (rogyar) [inline app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertProductsQtyForAttributeInLayeredNavigationActionGroup.xml:11]: Please, rename as `AssertStorefrontProductsCountForAttributeInLayeredNavigationActionGroup` and check my previous comment regarding the `qty` keyword. 

Thank you.

File: app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledByMassactionProductsTest.xml [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            <title value="Check configurable products count with disabled by massaction products"/>
+            <description value="Storefront Layered navigation with disabled by massaction products configurable children"/>
+            <severity value="MAJOR"/>
+            <testCaseId value="MAGETWO-90123"/>
+            <group value="layeredNavigation"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <!-- Create four categories assigned in ascending order:
```
Comment (rogyar) [inline app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledByMassactionProductsTest.xml:25]: Please, remove comments from the test. They might be considered as separate test steps in CI system.

File: app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest.xml [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest">
```
Comment (rogyar) [inline app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest.xml:11]: Please, correct the area name. It applies to all other places as well

```suggestion
    <test name="StorefrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest">
```

Thank you.

Comment (rogyar) [reply app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest.xml:11]: So, it's not `StoreFront` but `Storefront`

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertCategoryIsEmptyActionGroup.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertCategoryIsEmptyActionGroup.xml]: Fixed,  thank you.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontGoToSubCategory2PageActionGroup.xml (hunk 2 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontGoToSubCategory2PageActionGroup.xml]: Navigation approach has been changed as per recommendation.

File: app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertAttributesQtyForShoppingOptionInLayeredNavigationActionGroup.xml (hunk 3 above)
Comment (AnnaAPak) [reply app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/StoreFrontAssertAttributesQtyForShoppingOptionInLayeredNavigationActionGroup.xml]: Fixed, thanks.

File: app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledByMassactionProductsTest.xml (hunk 5 above)
Comment (AnnaAPak) [reply app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledByMassactionProductsTest.xml]: All comments have been removed.

File: app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest.xml (hunk 6 above)
Comment (AnnaAPak) [reply app/code/Magento/LayeredNavigation/Test/Mftf/Test/StoreFrontProductsCountInLayeredNavigationWithDisabledFromProductEditPageProductsTest.xml]: Addressed, thanks.

File: app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/AssertStorefrontProductsCountForAttributeInLayeredNavigationActionGroup.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            <argument name="count" type="string"/>
+            <argument name="attributeName" type="string"/>
+        </arguments>
+
+        <grabTextFrom selector="{{StorefrontLayeredNavigationSection.productsForAttribute(attributeName)}}" stepKey="grabProductCount"/>
+
+        <assertEquals stepKey="assertCount">
+           <actualResult type="const">$grabProductCount</actualResult>
+           <expectedResult type="string">{{count}}</expectedResult>
+         </assertEquals>
```
Comment (rogyar) [inline app/code/Magento/LayeredNavigation/Test/Mftf/ActionGroup/AssertStorefrontProductsCountForAttributeInLayeredNavigationActionGroup.xml:25]: Please, fix the indentation

**Pull request merged**